Source/Core/BingMapsGeocoderService.js

```diff
@@ -0,0 +1,113 @@
+/*global define*/
+define([
+    './BingMapsApi',
+    './defaultValue',
+    './defineProperties',
+    './loadJsonp',
+    './Rectangle',
+], function(
+    BingMapsApi,
+    defaultValue,
+    defineProperties,
+    loadJsonp,
+    Rectangle) {
+    'use strict';
+
+   var url = 'https://dev.virtualearth.net/REST/v1/Locations';
+
+    /**
+     * Provides geocoding through Bing Maps.
+     * @alias BingMapsGeocoderService
+     * @constructor
+     *
+     * @param {Object} options Object with the following properties:
+     * @param {String} [key] A key to use with the Bing Maps geocoding service
+     * @param {Boolean} autoComplete Indicates whether this service shall be used to fetch auto-complete suggestions
+     */
+    function BingMapsGeocoderService(options) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+        this._canceled = false;
+        this._url = 'https://dev.virtualearth.net/REST/v1/Locations';
+        this._key = BingMapsApi.getKey(options.key);
+
+        /**
+         * Indicates whether this geocoding service is to be used for autocomplete.
+         *
+         * @type {boolean}
+         * @default false
+         */
+        this.autoComplete = defaultValue(options.autoComplete, false);
```

Comment:
Why is this a Geocoder option instead of a widget-level option?  It seems odd to make the decision at the low-level, non-UX part.